x-pack/plugins/ml/server/lib/alerts/alerting_service.ts

```diff
@@ -257,6 +263,26 @@ export function alertingServiceProvider(mlClient: MlClient) {
     };
   };
 
+  /**
+   * Provides unique key for the anomaly result.
+   */
+  const getAlertInstanceKey = (source: any): string => {
+    let alertInstanceKey = `${source.job_id}_${source.timestamp}`;
+    if (source.result_type === ANOMALY_RESULT_TYPE.INFLUENCER) {
+      alertInstanceKey += `_${source.influencer_field_name}_${source.influencer_field_value}`;
+    } else if (source.result_type === ANOMALY_RESULT_TYPE.RECORD) {
+      const fieldName =
+        source.field_name ??
```

Comment:
I don't think this should be using `source.field_name` here - that is the name of the metric field, so you end up with something like `high_sum_taxful_total_price_Jim Stewart` in the key. Can you use `getEntityFieldName` and `getEntityFieldValue` from `common/util/anomaly_utils.ts` here instead?

Reply:
Changed in https://github.com/elastic/kibana/pull/91485/commits/a3d3d13b225dfc80c611d28d16b9054d07a7f4e1

Reply:
And extended with `detector_index` in https://github.com/elastic/kibana/pull/91485/commits/3e910c238ae187850f81b2be1b7cdff3cbdb9a3a